nilearn/decoding/tests/test_decoder.py

```diff
@@ -1144,6 +1156,40 @@ def test_decoder_screening_percentile_surface(perc, _make_surface_class_data):
         assert model.screening_percentile_ == perc
 
 
+@pytest.mark.parametrize("screening_percentile", [30, 80, 100])
+def test_decoder_screening_percentile_adjustment_surface(
+    _make_surface_mask, _make_surface_class_data, screening_percentile
+):
+    """Test adjustment of screening percentile, such that:
+    - if mask size is smaller than screening percentile wrt to the mesh size,
+    use whole mask.
+    - if mask size is greater than screening percentile wrt to the mesh size,
+    adjust the screening percentile for that mask size.
+    """
+    mask = _make_surface_mask()
+    img, y = _make_surface_class_data()
+    mask_n_vertices = get_mask_volume(mask)
```

Comment:
```
    mask_n_vertices = get_mask_volume(mask)
```

we will have to rename this private function or split its behavior because something called get_mask_volume that returns a number of vertices is confusing

Reply:
Ok I will create separate issues for this